sycl/include/CL/sycl/detail/kernel_program_cache.hpp

```diff
@@ -25,29 +25,42 @@ namespace detail {
 class context_impl;
 class KernelProgramCache {
 public:
-  /// Denotes pointer to some entity with its state.
+  /// Denotes build error data. The data is filled in from cl::sycl::exception
+  /// class instance.
+  struct BuildError {
+    std::string Msg;
+    pi_int32 Code;
+
+    /// Equals to true if the Msg and Code are initialized. This flag is added
+    /// due to possibility of Code equal to zero even if there is a
+    /// cl::sycl::exception thrown
```

Comment:
```suggestion
    /// due to the possibility of error code being equal to zero even in case
    /// if build is failed and an exception is thrown
```

Reply:
Done